google-cloud-firestore/src/main/java/com/google/cloud/firestore/FirestoreImpl.java

```diff
@@ -212,21 +220,35 @@ void getAll(
       @Nullable FieldMask fieldMask,
       @Nullable ByteString transactionId,
       final ApiStreamObserver<DocumentSnapshot> apiStreamObserver) {
-
     ResponseObserver<BatchGetDocumentsResponse> responseObserver =
         new ResponseObserver<BatchGetDocumentsResponse>() {
           int numResponses = 0;
           boolean hasCompleted = false;
 
           @Override
-          public void onStart(StreamController streamController) {}
+          public void onStart(StreamController streamController) {
+            getTraceUtil()
+                .currentSpan()
+                .addEvent(
+                    TraceUtil.SPAN_NAME_BATCH_GET_DOCUMENTS + ": Start",
+                    Collections.singletonMap("numDocuments", documentReferences.length));
+          }
 
           @Override
           public void onResponse(BatchGetDocumentsResponse response) {
             DocumentReference documentReference;
             DocumentSnapshot documentSnapshot;
 
             numResponses++;
+            if (numResponses == 1) {
+              getTraceUtil()
+                  .currentSpan()
+                  .addEvent(TraceUtil.SPAN_NAME_BATCH_GET_DOCUMENTS + ": First response received");
+            } else if (numResponses % 100 == 0) {
+              getTraceUtil()
+                  .currentSpan()
+                  .addEvent(TraceUtil.SPAN_NAME_BATCH_GET_DOCUMENTS + ": Received 100 responses");
```

Comment:
just curious, why is 100 treated specially here? Can you add a comment?

Reply:
This was done to make the trace size smaller. So for example if a query results in 1000 documents, instead of having 1000 "response received" events, we'll just have 10 events (one for every 100 responses).

This was done in the past with [opencensus traces](https://github.com/googleapis/java-firestore/blob/main/google-cloud-firestore/src/main/java/com/google/cloud/firestore/FirestoreImpl.java#L241-L244), so I decided to do the same thing for open-telemetry traces.

Added comments.